### ISSUE TITLE
[Doc] Java Execution requires more stack for very large configs 

### ISSUE THREAD

**Issue by karenzone** [elastic/logstash#11842]
Update documentation to note that java execution engine requires more stack for large configs.

* How can we quantify "large configs" and "more stack" into meaningful recommendations?
* If we can't provide exact numbers, what kind of heuristics can we provide?

**Comment by colinsurprenant** [comment]
Good questions and AFAIK there are no rules of thumb. Maybe just say that if a stack overflow error occurs, which is not uncommon with complex configurations, the JVM stack size can be increased using `-Xss4M` for example. 

One problem is that de *default* stack size is different per platform and per OS flavor. One way to know what the default is is by runinng
```sh
$ java -XX:+PrintFlagsFinal -version | grep ThreadStackSize
```

So I would say that depending on the default stack size I would start by multiplying by 4x, then 8x then 16x until it solves.

**Comment by karenzone** [comment]
@colinsurprenant Those sound like some pretty good rules of thumb to me. 😄   
Thank you for this info.

**Comment by colinsurprenant** [comment]
can we close this?

**Comment by karenzone** [comment]
Closed with #11867
